### PR TITLE
Fix nuget config not being honored

### PR DESCRIPTION
**Pull request by wli3**
Copy user level nuget config to override folder

Tested end to end locally. Pretty hard to add unit test for permission code

### CODE REVIEW

File: src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs [hunk 1]
```diff
@@ -33,7 +37,57 @@ public static void OverrideEnvironmentVariableToTmp(ParseResult parseResult)
             if (!OperatingSystem.IsWindows() && IsRunningUnderSudo() && IsRunningWorkloadCommand(parseResult))
             {
                 Directory.CreateDirectory(SudoHomeDirectory);
+
+                var homeBeforeOverride = Path.Combine(Environment.GetEnvironmentVariable("HOME"));
                 Environment.SetEnvironmentVariable("HOME", SudoHomeDirectory);
+
+                CopyUserNuGetConfigToOverridenHome(homeBeforeOverride);
```
Comment (KalleOlaviNiemitalo) [inline src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs:44]: The spelling is "[overridden](https://en.wiktionary.org/wiki/overridden#English)".

**Review by KalleOlaviNiemitalo — COMMENTED** [review]
I feel suspicious about the hardcoded `/tmp/dotnet_sudo_home/` — what if that directory has been maliciously created by a different, unprivileged user, who still has write access — but perhaps that was discussed in another PR.

File: src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs [hunk 2]
```diff
@@ -33,7 +37,57 @@ public static void OverrideEnvironmentVariableToTmp(ParseResult parseResult)
             if (!OperatingSystem.IsWindows() && IsRunningUnderSudo() && IsRunningWorkloadCommand(parseResult))
             {
                 Directory.CreateDirectory(SudoHomeDirectory);
+
+                var homeBeforeOverride = Path.Combine(Environment.GetEnvironmentVariable("HOME"));
```
Comment (sfoslund) [inline src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs:41]: Sorry, I must have missed this in the first check in of this change, but directly using the HOME env var doesn't match the logic we use for calculating the home dir here: https://github.com/dotnet/sdk/blob/release/6.0.1xx/src/Common/CliFolderPathCalculatorCore.cs#L35, should we be taking those other env vars into account?

Comment (wli3) [reply src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs:41]: That's fine. If the user override the home directory, they are in control of the situation. We just need to make sure the default make sense.

**Comment by wli3** [comment]
@sfoslund looks good?

**Review by sfoslund — APPROVED** [review]

**Comment by wli3** [comment]
@KalleOlaviNiemitalo thank you. I added logic to ensure the owner is root

**Comment by KalleOlaviNiemitalo** [comment]
@wli3 `lstat` might be more secure for that; wouldn't be tricked by a user-owned symlink to a root-owned directory. But I did not check whether you already prevent that in some other way.

File: src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs [hunk 3]
```diff
@@ -32,12 +37,93 @@ public static void OverrideEnvironmentVariableToTmp(ParseResult parseResult)
...
+
+        private static bool IsOtherUserWritable(StatInterop.FileStatus fileStat)
+        {
+            return (fileStat.Mode & (int) StatInterop.Permissions.S_IWOTH) == 0;
+        }
+
+        private static bool IsGroupWritable(StatInterop.FileStatus fileStat)
+        {
+            return (fileStat.Mode & (int) StatInterop.Permissions.S_IWGRP) == 0;
+        }
```
[the remark this answers is not in the document]
Comment (wli3) [reply src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs:122]: could you give me more detail? Any suggestions?

**Comment by wli3** [comment]
@joeloff @sfoslund could you both have a look again? especially on https://github.com/dotnet/sdk/pull/20930/commits/d239c27aac0fc61380ad8f408ad757de286467c3

File: src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (wli3) [reply src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs]: @joeloff @KalleOlaviNiemitalo now I get it. Rename to "...CannotWrite"

**Review by joeloff — APPROVED** [review]

**Pull request merged**